src/tests/io/test_loadtxt.f90

```diff
@@ -1,55 +1,63 @@
 program test_loadtxt
-use stdlib_experimental_kinds, only: int32, sp, dp
-use stdlib_experimental_io, only: loadtxt
-use stdlib_experimental_error, only: error_stop
-implicit none
+  use stdlib_experimental_kinds, only: int32, sp, dp
+  use stdlib_experimental_io, only: loadtxt
+  use stdlib_experimental_error, only: error_stop
+  implicit none
 
-integer(int32), allocatable :: i(:, :)
-real(sp), allocatable :: s(:, :)
-real(dp), allocatable :: d(:, :)
+  integer(int32), allocatable :: i(:, :)
+  real(sp), allocatable :: s(:, :)
+  real(dp), allocatable :: d(:, :)
+  complex(dp), allocatable :: z(:, :)
```

Comment:
```suggestion
  complex(sp), allocatable :: c(:, :)
  complex(dp), allocatable :: z(:, :)
```
Since you implemented `loadtxt` for `complex(sp)`, shouldn't we test it as well?